lib/tuntap.mli

```diff
@@ -60,15 +60,19 @@ val set_up_and_running : string -> unit
     running. Note that when using the [set_ipv4] function, the
     interface will automatically be set up and running. *)
 
-type ipaddr =
-  | AF_INET of Ipaddr.V4.t * Ipaddr.V4.Prefix.t
-  | AF_INET6 of Ipaddr.V6.t * Ipaddr.V6.Prefix.t
+val getifaddrs : ?wanted:([`V4 | `V6] list) -> unit ->
+  [ `V4 of string * Ipaddr.V4.t * Ipaddr.V4.Prefix.t
+  | `V6 of string * Ipaddr.V6.t * Ipaddr.V6.Prefix.t ] list
+(** [getifaddrs ()] is the list that associates interface names with
+    their available IP addresses and their prefix. *)
```

Comment:
On 08/01/2015 19:45, Richard Mortier wrote:
> can we not have interfaces that have not yet acquired either an IP
> address of either type?

This function is a wrapper to getifaddrs(3). I did expose just the bit
of functionality we were interested in: obtain the address of an
interface. Sure, this could be better bound to the original C function.
I did not want to spend much on other address types (like IPX, etc.
etc.) because we (almost?) never use them.

> does this return for all interfaces? 

Not sure what you mean. It returns a list of interface that possess at
least an IP address. Other interfaces are not returned.

wouldn't this be better named
> |getallifaddrs| or something? 

The getifaddrs(3) C function already returns a linked list of
interfaces, since it is similar here, I kept the same name.

and perhaps provide a |val getifaddrs:
> ?wanted:([`V4 | `V6] list) -> string -> unit -> [ ... ] list| as well?

getifaddr without *s* then. Sure, I'm doing this, this would be useful
indeed.

-- 
Vincent